File: lib/src/screens/ride_screen.dart
import 'package:flutter/material.dart';
import '../constants/app_color.dart';
import '../models/ride.dart';
import '../models/ride_status.dart';
import '../widgets/ride_card.dart';
import '../widgets/components/modern_card.dart';
import 'package:intl/intl.dart';

/// Screen that displays a list of upcoming and past rides
class RideScreen extends StatefulWidget {
  /// Creates a ride screen
  const RideScreen({super.key});

  @override
  State<RideScreen> createState() => _RideScreenState();
}

class _RideScreenState extends State<RideScreen>
    with SingleTickerProviderStateMixin {
  late TabController _tabController;
  bool _showContent = false;

  // Mock data for testing
  List<Ride> upcomingRides = [];
  List<Ride> pastRides = [];

  @override
  void initState() {
    super.initState();
    _tabController = TabController(length: 2, vsync: this);
    _loadRideData();

    // Start animations
    Future.delayed(const Duration(milliseconds: 300), () {
      if (mounted) {
        setState(() {
          _showContent = true;
        });
      }
    });
  }

  void _loadRideData() {
    // Mock data
    upcomingRides = [
      Ride(
        id: 'ride1',
        userId: 'test_id',
        pickupAddress: '123 Home St, Anytown',
        destinationAddress: '456 Hospital Ave, Medville',
        dateTime: DateTime.now().add(const Duration(days: 1)),
        status: RideStatus.scheduled,
        fare: 25.0,
        distance: 12.5,
        duration: 25,
        passengerName: 'John Doe',
        rideType: 'AMBULATORY',
        driverName: 'Michael Johnson',
        vehicleInfo: 'Toyota Camry - White (ABC123)',
      ),
      Ride(
        id: 'ride2',
        userId: 'test_id',
        pickupAddress: '123 Home St, Anytown',
        destinationAddress: '789 Clinic Rd, Healtown',
        dateTime: DateTime.now().add(const Duration(days: 5)),
        status: RideStatus.scheduled,
        fare: 30.0,
        distance: 15.0,
        duration: 35,
        passengerName: 'John Doe',
        rideType: 'AMBULATORY',
      ),
    ];

    pastRides = [
      Ride(
        id: 'ride3',
        userId: 'test_id',
        pickupAddress: '123 Home St, Anytown',
        destinationAddress: '456 Hospital Ave, Medville',
        dateTime: DateTime.now().subtract(const Duration(days: 1)),
        status: RideStatus.completed,
        fare: 28.0,
        distance: 13.0,
        duration: 30,
        passengerName: 'John Doe',
        rideType: 'AMBULATORY',
        driverName: 'Sarah Williams',
        vehicleInfo: 'Honda Accord - Silver (XYZ789)',
        actualPrice: 32.75,
      ),
      Ride(
        id: 'ride4',
        userId: 'test_id',
        pickupAddress: '123 Home St, Anytown',
        destinationAddress: '321 Therapy Center, Welltown',
        dateTime: DateTime.now().subtract(const Duration(days: 5)),
        status: RideStatus.cancelled,
        fare: 35.0,
        distance: 18.0,
        duration: 40,
        passengerName: 'John Doe',
        rideType: 'AMBULATORY',
      ),
    ];
  }

  @override
  void dispose() {
    _tabController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text(
          'My Rides',
          style: TextStyle(
            fontSize: 24,
            fontWeight: FontWeight.w600,
            color: AppColor.textDarkBlue,
          ),
        ),
        backgroundColor: Colors.transparent,
        elevation: 0,
        iconTheme: const IconThemeData(color: AppColor.textDarkBlue),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back),
          onPressed: () {
            // Navigate back to the modern home screen
            Navigator.of(context).pushReplacementNamed('/home');
          },
        ),
        actions: [
          IconButton(
            icon: const Icon(Icons.filter_alt_outlined),
            onPressed: () {
              // Filter functionality
            },
          ),
        ],
      ),
      extendBodyBehindAppBar: true,
      body: Container(
        decoration: const BoxDecoration(
          gradient: LinearGradient(
            begin: Alignment.topCenter,
            end: Alignment.bottomCenter,
            colors: [
              Color(0xFFF8FAFC),
              Color(0xFFF1F5FD),
            ],
          ),
        ),
        child: SafeArea(
          child: Column(
            children: [
              // Custom Tabs
              _buildCustomTabs(),

              // Ride list
              Expanded(
                child: AnimatedOpacity(
                  opacity: _showContent ? 1.0 : 0.0,
                  duration: const Duration(milliseconds: 500),
                  curve: Curves.easeInOut,
                  child: AnimatedSlide(
                    offset: _showContent ? Offset.zero : const Offset(0, 0.05),
                    duration: const Duration(milliseconds: 500),
                    curve: Curves.easeOutCubic,
                    child: TabBarView(
                      controller: _tabController,
                      children: [
                        // Upcoming rides tab
                        _buildRideList(upcomingRides, "No upcoming rides"),

                        // Past rides tab
                        _buildRideList(pastRides, "No past rides"),
                      ],
                    ),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildCustomTabs() {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 24.0, vertical: 16.0),
      child: Container(
        height: 48,
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(50),
          color: Colors.white,
          border: Border.all(color: Colors.grey.shade300),
          boxShadow: [
            BoxShadow(
              color: Colors.black.withOpacity(0.05),
              blurRadius: 4,
              offset: const Offset(0, 2),
            ),
          ],
        ),
        child: TabBar(
          controller: _tabController,
          indicator: BoxDecoration(
            borderRadius: BorderRadius.circular(50),
            gradient: const LinearGradient(
              colors: [
                AppColor.primaryBlue,
                AppColor.primaryPurple,
              ],
            ),
          ),
          // Make the tab indicator take up full height of the tab bar
          indicatorSize: TabBarIndicatorSize.tab,
          // No padding around the tab indicator
          padding: EdgeInsets.zero,
          labelPadding: EdgeInsets.zero,
          // Use a custom divider decoration
          dividerColor: Colors.transparent,
          // Text styling
          labelColor: Colors.white,
          unselectedLabelColor: AppColor.textDarkBlue,
          labelStyle: const TextStyle(
            fontWeight: FontWeight.w600,
            fontSize: 16,
          ),
          unselectedLabelStyle: const TextStyle(
            fontWeight: FontWeight.w400,
            fontSize: 16,
          ),
          tabs: const [
            Tab(
              text: 'Upcoming',
              height: 48,
            ),
            Tab(
              text: 'Past',
              height: 48,
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildRideList(List<Ride> rides, String emptyMessage) {
    if (rides.isEmpty) {
      return _buildEmptyState(emptyMessage);
    } else {
      return ListView.builder(
        padding: const EdgeInsets.all(16),
        itemCount: rides.length,
        itemBuilder: (context, index) {
          final ride = rides[index];

          // Use a TweenAnimationBuilder to create a staggered animation effect
          return TweenAnimationBuilder<double>(
            tween: Tween(begin: 0.0, end: 1.0),
            duration: Duration(milliseconds: 300 + (index * 100)),
            builder: (context, value, child) {
              return Opacity(
                opacity: value,
                child: Transform.translate(
                  offset: Offset(0, 20 * (1 - value)),
                  child: child,
                ),
              );
            },
            child: Padding(
              padding: const EdgeInsets.only(bottom: 16.0),
              child: EnhancedRideCard(
                ride: ride,
                onTap: () {
                  // Navigate to ride details
                  Navigator.pushNamed(context, '/ride-details',
                      arguments: ride);
                },
              ),
            ),
          );
        },
      );
    }
  }

  Widget _buildEmptyState(String message) {
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          _buildEmptyStateIllustration(),
          const SizedBox(height: 24),
          Text(
            message,
            style: const TextStyle(
              fontSize: 18,
              fontWeight: FontWeight.w500,
              color: AppColor.textDarkBlue,
            ),
          ),
          const SizedBox(height: 8),
          const Text(
            'Your rides will appear here once scheduled',
            style: TextStyle(
              fontSize: 14,
              color: AppColor.textMediumGray,
            ),
            textAlign: TextAlign.center,
          ),
          const SizedBox(height: 24),
          ElevatedButton.icon(
            onPressed: () {
              // Book a ride
            },
            icon: const Icon(Icons.add, size: 20),
            label: const Text(
              'Book a Ride',
              style: TextStyle(fontWeight: FontWeight.w500),
            ),
            style: ElevatedButton.styleFrom(
              backgroundColor: AppColor.primaryBlue,
              foregroundColor: Colors.white,
              padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(50),
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildEmptyStateIllustration() {
    return SizedBox(
      height: 180,
      width: 180,
      child: Stack(
        alignment: Alignment.center,
        children: [
          // Background circle
          Container(
            width: 160,
            height: 160,
            decoration: BoxDecoration(
              shape: BoxShape.circle,
              color: AppColor.primaryPurple.withOpacity(0.1),
            ),
          ),

          // Simple car illustration
          TweenAnimationBuilder<double>(
            tween: Tween(begin: 0.8, end: 1.1),
            duration: const Duration(milliseconds: 1500),
            curve: Curves.easeInOut,
            builder: (context, value, child) {
              return Transform.scale(
                scale: value,
                child: child,
              );
            },
            child: const Icon(
              Icons.directions_car_outlined,
              size: 80,
              color: AppColor.primaryBlue,
            ),
          ),
        ],
      ),
    );
  }
}

/// Enhanced version of the ride card that more closely matches the Kotlin implementation
class EnhancedRideCard extends StatelessWidget {
  /// The ride to display
  final Ride ride;

  /// Callback when the card is tapped
  final VoidCallback onTap;

  /// Creates an enhanced ride card
  const EnhancedRideCard({
    super.key,
    required this.ride,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return ModernCard(
      onTap: onTap,
      elevation: 2,
      borderRadius: 24,
      padding: const EdgeInsets.all(20),
      child: Column(
        children: [
          // Top section with date, time and status
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              // Date with icon
              _buildIconPill(
                icon: Icons.calendar_today_outlined,
                iconColor: AppColor.primaryBlue,
                backgroundColor: AppColor.primaryBlue.withOpacity(0.1),
                text: _formatDate(ride.dateTime),
              ),

              // Time with icon
              _buildIconPill(
                icon: Icons.schedule_outlined,
                iconColor: AppColor.primaryPurple,
                backgroundColor: AppColor.primaryPurple.withOpacity(0.1),
                text: _formatTime(ride.dateTime),
              ),

              // Status badge
              StatusBadge(status: ride.status),
            ],
          ),

          const SizedBox(height: 16),

          // Divider with gradient
          Container(
            height: 1,
            width: double.infinity,
            decoration: BoxDecoration(
              gradient: LinearGradient(
                colors: [
                  AppColor.primaryBlue.withOpacity(0.1),
                  AppColor.primaryPurple.withOpacity(0.1),
                  AppColor.primaryBlue.withOpacity(0.1),
                ],
              ),
            ),
          ),

          const SizedBox(height: 16),

          // Route information with visual connector
          Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // Source and destination icons with connector
              Column(
                children: [
                  // Source dot
                  Container(
                    width: 12,
                    height: 12,
                    decoration: const BoxDecoration(
                      shape: BoxShape.circle,
                      color: AppColor.primaryBlue,
                    ),
                  ),

                  // Connector line
                  Container(
                    width: 2,
                    height: 40,
                    decoration: const BoxDecoration(
                      gradient: LinearGradient(
                        begin: Alignment.topCenter,
                        end: Alignment.bottomCenter,
                        colors: [
                          AppColor.primaryBlue,
                          AppColor.primaryPurple,
                        ],
                      ),
                    ),
                  ),

                  // Destination dot
                  Container(
                    width: 12,
                    height: 12,
                    decoration: const BoxDecoration(
                      shape: BoxShape.circle,
                      color: AppColor.primaryPurple,
                    ),
                  ),
                ],
              ),

              const SizedBox(width: 16),

              // Addresses
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    // Pickup address
                    Text(
                      ride.pickupAddress,
                      style: const TextStyle(
                        fontSize: 14,
                        color: AppColor.textDarkBlue,
                      ),
                    ),

                    const SizedBox(height: 40),

                    // Destination address
                    Text(
                      ride.destinationAddress,
                      style: const TextStyle(
                        fontSize: 14,
                        color: AppColor.textDarkBlue,
                      ),
                    ),
                  ],
                ),
              ),
            ],
          ),

          // Actions section for upcoming rides
          if (ride.status == RideStatus.scheduled) ...[
            const SizedBox(height: 16),
            Row(
              mainAxisAlignment: MainAxisAlignment.end,
              children: [
                TextButton.icon(
                  onPressed: () {
                    // Cancel ride
                  },
                  icon: const Icon(
                    Icons.close,
                    size: 16,
                    color: AppColor.errorRed,
                  ),
                  label: const Text(
                    'Cancel',
                    style: TextStyle(
                      fontWeight: FontWeight.w500,
                      color: AppColor.errorRed,
                    ),
                  ),
                ),
                TextButton.icon(
                  onPressed: onTap,
                  label: const Text(
                    'View Details',
                    style: TextStyle(
                      fontWeight: FontWeight.w500,
                      color: AppColor.primaryBlue,
                    ),
                  ),
                  icon: const Icon(
                    Icons.arrow_forward,
                    size: 16,
                    color: AppColor.primaryBlue,
                  ),
                ),
              ],
            ),
          ],
        ],
      ),
    );
  }

  Widget _buildIconPill({
    required IconData icon,
    required Color iconColor,
    required Color backgroundColor,
    required String text,
  }) {
    return Row(
      children: [
        Container(
          width: 32,
          height: 32,
          decoration: BoxDecoration(
            shape: BoxShape.circle,
            color: backgroundColor,
          ),
          child: Icon(
            icon,
            size: 16,
            color: iconColor,
          ),
        ),
        const SizedBox(width: 8),
        Text(
          text,
          style: const TextStyle(
            fontWeight: FontWeight.w500,
            fontSize: 14,
            color: AppColor.textDarkBlue,
          ),
        ),
      ],
    );
  }

  String _formatDate(DateTime dateTime) {
    final formatter = DateFormat('EEE, MMM d');
    return formatter.format(dateTime);
  }

  String _formatTime(DateTime dateTime) {
    final formatter = DateFormat('h:mm a');
    return formatter.format(dateTime);
  }
}

File: backup/kotlin/ui/RideScreen.kt
package com.example.healthride.ui

import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.outlined.*
import androidx.compose.material3.*
import androidx.compose.material3.TabRowDefaults.tabIndicatorOffset
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.*
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.example.healthride.data.model.Ride
import com.example.healthride.data.model.RideStatus
import com.example.healthride.SampleData
import com.example.healthride.formatToPattern
import com.example.healthride.ui.components.GradientBackground
import com.example.healthride.ui.components.GradientType
import com.example.healthride.ui.theme.*
import kotlinx.coroutines.delay
import java.util.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RidesScreen(
    onBackClick: () -> Unit = {},
    onRideClick: (String) -> Unit = {}
) {
    val tabTitles = listOf("Upcoming", "Past")
    var selectedTabIndex by remember { mutableStateOf(0) }

    // Animation states
    var showContent by remember { mutableStateOf(false) }

    // Use SampleData lists
    val upcomingRides = SampleData.upcomingRides
    val pastRides = SampleData.pastRides

    // Start animations
    LaunchedEffect(Unit) {
        delay(300) // Brief delay for startup animation
        showContent = true
    }

    GradientBackground(gradientType = GradientType.BLUE_LIGHT) {
        Scaffold(
            topBar = {
                TopAppBar(
                    title = { Text("My Rides", fontWeight = FontWeight.SemiBold) },
                    navigationIcon = {
                        IconButton(onClick = onBackClick) {
                            Icon(Icons.Default.ArrowBack, "Back")
                        }
                    },
                    actions = {
                        IconButton(onClick = { /* Filter */ }) {
                            Icon(Icons.Outlined.FilterAlt, "Filter")
                        }
                    },
                    colors = TopAppBarDefaults.topAppBarColors(
                        containerColor = Color.Transparent,
                        titleContentColor = TextDarkBlue
                    )
                )
            },
            containerColor = Color.Transparent // Make scaffold transparent to show gradient
        ) { innerPadding ->
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(innerPadding)
            ) {
                // Custom Animated Tabs
                CustomAnimatedTabs(
                    tabTitles = tabTitles,
                    selectedTabIndex = selectedTabIndex,
                    onTabSelected = { selectedTabIndex = it }
                )

                // Animated content
                AnimatedVisibility(
                    visible = showContent,
                    enter = fadeIn() + slideInVertically(initialOffsetY = { it / 3 })
                ) {
                    Box(modifier = Modifier.fillMaxSize()) {
                        // Determine which list to show
                        val currentList = if (selectedTabIndex == 0) upcomingRides else pastRides
                        val emptyMsg = if (selectedTabIndex == 0) "No upcoming rides" else "No past rides"

                        // Display the list or empty message
                        EnhancedRideList(
                            rides = currentList,
                            onRideClick = onRideClick,
                            emptyMessage = emptyMsg
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun CustomAnimatedTabs(
    tabTitles: List<String>,
    selectedTabIndex: Int,
    onTabSelected: (Int) -> Unit
) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 8.dp)
            .clip(RoundedCornerShape(24.dp))
            .border(
                width = 1.dp,
                color = Color(0xFFE0E0E0),
                shape = RoundedCornerShape(24.dp)
            )
            .background(SurfaceWhite.copy(alpha = 0.8f)),
    ) {
        TabRow(
            selectedTabIndex = selectedTabIndex,
            containerColor = Color.Transparent,
            contentColor = PrimaryBlue,
            indicator = { tabPositions ->
                // Custom animated indicator
                if (selectedTabIndex < tabPositions.size) {
                    Box(
                        Modifier
                            .tabIndicatorOffset(tabPositions[selectedTabIndex])
                            .fillMaxWidth(1f / tabPositions.size)
                            .padding(3.dp)
                            .height(36.dp)
                            .clip(RoundedCornerShape(18.dp))
                            .background(
                                brush = Brush.horizontalGradient(
                                    colors = listOf(
                                        PrimaryBlue.copy(alpha = 0.15f),
                                        PrimaryPurple.copy(alpha = 0.15f)
                                    )
                                )
                            )
                            .border(
                                width = 1.dp,
                                brush = Brush.horizontalGradient(
                                    colors = listOf(PrimaryBlue, PrimaryPurple)
                                ),
                                shape = RoundedCornerShape(18.dp)
                            )
                    )
                }
            },
            divider = {}
        ) {
            tabTitles.forEachIndexed { index, title ->
                Tab(
                    selected = selectedTabIndex == index,
                    onClick = { onTabSelected(index) },
                    text = {
                        Text(
                            text = title,
                            fontWeight = FontWeight.Medium,
                            color = if (selectedTabIndex == index) PrimaryBlue else TextMediumGray
                        )
                    },
                    modifier = Modifier.height(42.dp)
                )
            }
        }
    }
}

@Composable
fun EnhancedRideList(
    rides: List<Ride>,
    onRideClick: (String) -> Unit,
    emptyMessage: String
) {
    if (rides.isEmpty()) {
        // Enhanced empty state visualization
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(16.dp),
            contentAlignment = Alignment.Center
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center
            ) {
                // Empty state illustration
                EmptyStateIllustration()

                Spacer(modifier = Modifier.height(24.dp))

                Text(
                    text = emptyMessage,
                    style = MaterialTheme.typography.titleMedium,
                    color = TextMediumGray,
                    textAlign = TextAlign.Center
                )

                Spacer(modifier = Modifier.height(8.dp))

                Text(
                    text = "Your rides will appear here once scheduled",
                    style = MaterialTheme.typography.bodyMedium,
                    color = TextMediumGray,
                    textAlign = TextAlign.Center
                )

                Spacer(modifier = Modifier.height(24.dp))

                Button(
                    onClick = { /* Book now functionality */ },
                    shape = RoundedCornerShape(50.dp),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = PrimaryBlue
                    )
                ) {
                    Icon(
                        Icons.Outlined.Add,
                        contentDescription = null,
                        modifier = Modifier.size(20.dp)
                    )

                    Spacer(modifier = Modifier.width(8.dp))

                    Text(
                        "Book a Ride",
                        fontWeight = FontWeight.Medium
                    )
                }
            }
        }
    } else {
        // Enhanced ride list with staggered animation
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
            contentPadding = PaddingValues(vertical = 16.dp)
        ) {
            items(rides, key = { ride -> ride.id }) { ride ->
                // Staggered animation for each item
                var itemVisible by remember { mutableStateOf(false) }

                LaunchedEffect(ride.id) {
                    delay(rides.indexOf(ride) * 100L) // Staggered delay
                    itemVisible = true
                }

                AnimatedVisibility(
                    visible = itemVisible,
                    enter = fadeIn() + slideInVertically { it / 4 }
                ) {
                    EnhancedRideCard(
                        ride = ride,
                        onClick = { onRideClick(ride.id) }
                    )
                }
            }
        }
    }
}

@Composable
fun EnhancedRideCard(
    ride: Ride,
    onClick: () -> Unit
) {
    val cardShape = RoundedCornerShape(24.dp)

    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick),
        shape = cardShape,
        colors = CardDefaults.cardColors(
            containerColor = SurfaceWhite
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 2.dp
        )
    ) {
        Column(modifier = Modifier.padding(20.dp)) {
            // Top section with date and status
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                // Date with icon
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        modifier = Modifier
                            .size(32.dp)
                            .clip(CircleShape)
                            .background(PrimaryBlueLight),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            Icons.Outlined.CalendarToday,
                            contentDescription = null,
                            tint = PrimaryBlue,
                            modifier = Modifier.size(16.dp)
                        )
                    }

                    Spacer(modifier = Modifier.width(8.dp))

                    Text(
                        text = ride.dateTime.formatToPattern("EEE, MMM d"),
                        fontWeight = FontWeight.Medium,
                        color = TextDarkBlue
                    )
                }

                // Time with icon
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        modifier = Modifier
                            .size(32.dp)
                            .clip(CircleShape)
                            .background(PrimaryPurpleLight),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            Icons.Outlined.Schedule,
                            contentDescription = null,
                            tint = PrimaryPurple,
                            modifier = Modifier.size(16.dp)
                        )
                    }

                    Spacer(modifier = Modifier.width(8.dp))

                    Text(
                        text = ride.dateTime.formatToPattern("h:mm a"),
                        fontWeight = FontWeight.Medium,
                        color = TextDarkBlue
                    )
                }

                // Status badge
                StatusBadge(status = ride.status)
            }

            Spacer(modifier = Modifier.height(16.dp))

            // Divider with gradient
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(1.dp)
                    .background(
                        brush = Brush.horizontalGradient(
                            colors = listOf(
                                PrimaryBlue.copy(alpha = 0.1f),
                                PrimaryPurple.copy(alpha = 0.1f),
                                PrimaryBlue.copy(alpha = 0.1f)
                            )
                        )
                    )
            )

            Spacer(modifier = Modifier.height(16.dp))

            // Route info with visual connector
            Row(
                modifier = Modifier.fillMaxWidth()
            ) {
                // Source and destination icons with connector
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    // Source dot
                    Box(
                        modifier = Modifier
                            .size(12.dp)
                            .clip(CircleShape)
                            .background(PrimaryBlue)
                    )

                    // Connector line
                    Box(
                        modifier = Modifier
                            .width(2.dp)
                            .height(40.dp)
                            .background(
                                brush = Brush.verticalGradient(
                                    colors = listOf(
                                        PrimaryBlue, PrimaryPurple
                                    )
                                )
                            )
                    )

                    // Destination dot
                    Box(
                        modifier = Modifier
                            .size(12.dp)
                            .clip(CircleShape)
                            .background(PrimaryPurple)
                    )
                }

                Spacer(modifier = Modifier.width(16.dp))

                // Addresses
                Column(
                    verticalArrangement = Arrangement.SpaceBetween,
                    modifier = Modifier.height(70.dp)
                ) {
                    // Pickup address
                    Text(
                        text = ride.pickupAddress,
                        style = MaterialTheme.typography.bodyMedium,
                        color = TextDarkBlue
                    )

                    // Destination address
                    Text(
                        text = ride.destinationAddress,
                        style = MaterialTheme.typography.bodyMedium,
                        color = TextDarkBlue
                    )
                }
            }

            // Actions section for upcoming rides
            if (ride.status == RideStatus.SCHEDULED) {
                Spacer(modifier = Modifier.height(16.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.End
                ) {
                    TextButton(
                        onClick = { /* Cancel ride */ },
                        colors = ButtonDefaults.textButtonColors(
                            contentColor = ErrorRed
                        )
                    ) {
                        Icon(
                            Icons.Outlined.Close,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp)
                        )

                        Spacer(modifier = Modifier.width(4.dp))

                        Text(
                            "Cancel",
                            fontWeight = FontWeight.Medium
                        )
                    }

                    TextButton(
                        onClick = { onClick() },
                        colors = ButtonDefaults.textButtonColors(
                            contentColor = PrimaryBlue
                        )
                    ) {
                        Text(
                            "View Details",
                            fontWeight = FontWeight.Medium
                        )

                        Spacer(modifier = Modifier.width(4.dp))

                        Icon(
                            Icons.Outlined.ArrowForward,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun StatusBadge(status: RideStatus) {
    val (backgroundColor, textColor, text) = when (status) {
        RideStatus.SCHEDULED -> Triple(PrimaryBlue.copy(alpha = 0.1f), PrimaryBlue, "Scheduled")
        RideStatus.IN_PROGRESS -> Triple(WarningYellow.copy(alpha = 0.1f), WarningYellow, "In Progress")
        RideStatus.COMPLETED -> Triple(SuccessGreen.copy(alpha = 0.1f), SuccessGreen, "Completed")
        RideStatus.CANCELLED -> Triple(ErrorRed.copy(alpha = 0.1f), ErrorRed, "Cancelled")
    }

    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(50.dp))
            .background(backgroundColor)
            .padding(horizontal = 12.dp, vertical = 6.dp)
    ) {
        Text(
            text = text,
            color = textColor,
            fontWeight = FontWeight.Medium,
            style = MaterialTheme.typography.labelSmall
        )
    }
}

@Composable
fun EmptyStateIllustration() {
    val infiniteTransition = rememberInfiniteTransition(label = "empty_state_animation")
    val dotScale = infiniteTransition.animateFloat(
        initialValue = 0.8f,
        targetValue = 1.2f,
        animationSpec = infiniteRepeatable(
            animation = tween(800, easing = EaseInOut),
            repeatMode = RepeatMode.Reverse
        ),
        label = "dot_scale"
    )

    Canvas(
        modifier = Modifier
            .size(180.dp)
            .padding(16.dp)
    ) {
        // Background circle
        drawCircle(
            color = PrimaryPurpleLight.copy(alpha = 0.3f),
            radius = size.minDimension / 2.2f,
            center = center
        )

        // Car or transportation icon (simplified)
        // Center coordinates
        val cx = center.x
        val cy = center.y
        val size = size.minDimension * 0.4f

        // Car body
        drawRoundRect(
            color = PrimaryBlue,
            topLeft = androidx.compose.ui.geometry.Offset(cx - size / 2, cy - size / 4),
            size = androidx.compose.ui.geometry.Size(size, size / 2),
            cornerRadius = androidx.compose.ui.geometry.CornerRadius(10f)
        )

        // Car roof
        drawRoundRect(
            color = PrimaryPurple,
            topLeft = androidx.compose.ui.geometry.Offset(cx - size / 3, cy - size / 2),
            size = androidx.compose.ui.geometry.Size(size * 2 / 3, size / 4),
            cornerRadius = androidx.compose.ui.geometry.CornerRadius(10f)
        )

        // Wheels
        val wheelRadius = size / 8 * dotScale.value
        drawCircle(
            color = TextDarkBlue,
            radius = wheelRadius,
            center = androidx.compose.ui.geometry.Offset(cx - size / 3, cy + size / 4)
        )
        drawCircle(
            color = TextDarkBlue,
            radius = wheelRadius,
            center = androidx.compose.ui.geometry.Offset(cx + size / 3, cy + size / 4)
        )
    }
}